src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1459,10 +1461,19 @@ private void AssignPatternVariables(BoundPattern pattern)
                         break;
                     }
                 case BoundKind.NegatedPattern:
+                    {
+                        var pat = (BoundNegatedPattern)pattern;
+                        AssignPatternVariables(pat.Negated, definitely: false);
```

Comment:
are there tests that verify diagnostics, or verify definite assignment state, on such pattern variables (contained within 'not' patterns)?

Reply:
See `PatternMatchingTests3.DisallowDesignatorsUnderNotAndOr`

---
In reply to: [380984943](https://github.com/dotnet/roslyn/pull/41503#discussion_r380984943) [](ancestors = 380984943)